pydough/qdag/collections/sub_collection.py

```diff
@@ -51,7 +52,14 @@ def is_singular(self, context: PyDoughCollectionQDAG) -> bool:
         relative_ancestor: PyDoughCollectionQDAG = (
             self.ancestor_context.starting_predecessor
         )
-        return (context == relative_ancestor) or relative_ancestor.is_singular(context)
+        return (
+            (context == relative_ancestor)
+            or relative_ancestor.is_singular(context)
+            or (
+                isinstance(self.ancestor_context, Singular)
+                and self.ancestor_context.is_singular(context)
```

Comment:
This is needed because, we would want to make the term from the top till the preceding Singular node, singular. Note that the, singular check between ancestor and current subcollection access is done before at Lines 50-51.